superset/views/csv_import.py

```diff
@@ -168,56 +179,82 @@ def add(self) -> Response:
-                db_name
-            )
-            raise ValueError(message)
-        try:
-            item = SQLAInterface(models.Database).obj()
-            item.database_name = db_name
-            item.sqlalchemy_uri = "sqlite:///" + db_path
-            item.allow_csv_upload = True
-            # TODO check if SQL-injection is possible through add()
-            db.session.add(item)
-            db.session.commit()
+        if db_flavor == "postgres":
+            # TODO add possibility to change user
+            # TODO add possibility to use schema
+            if not password:
+                raise NoPasswordSuppliedException
+            url = "postgresql://postgres:" + password + "@localhost/" + db_name
+            engine = sqlalchemy.create_engine(url)
+
+            # TODO decide whether to fail if database exists
+            try:
+                item = SQLAInterface(models.Database).obj()
+                item.database_name = db_name
+                item.sqlalchemy_uri = repr(engine.url)
+                item.password = password
+                item.allow_csv_upload = True
+                db.session.add(item)
+                db.session.commit()
+            except Exception as e:
+                stats_logger.incr("failed_csv_upload")
+                raise e
+
             return item
-        except Exception as e:
-            exception = e
-            if isinstance(e, IntegrityError):
-                message = "Error when trying to create Database. A database with the name {0} already exists.".format(
+        else:
+            db_path = os.getcwd() + "/" + db_name + ".db"
+            if os.path.isfile(db_path):
+                message = "Database file for {0} already exists, please choose a different name".format(
                     db_name
                 )
-                exception = DatabaseCreationException(message)
+                raise ValueError(message)
             try:
-                if os.path.isfile(db_path):
-                    os.remove(db_path)
-                db.session.delete(item)
+                item = SQLAInterface(models.Database).obj()
+                item.database_name = db_name
+                item.sqlalchemy_uri = "sqlite:///" + db_path
+                item.allow_csv_upload = True
+                # TODO check if SQL-injection is possible through add()
+                db.session.add(item)
                 db.session.commit()
-            except OSError:
-                message = _(
-                    "Error when trying to create Database.The database file {0}.db could not be removed. "
-                    "Please contact your administrator to remove it manually".format(
+                return item
+            except Exception as e:
+                exception = e
+                if isinstance(e, IntegrityError):
+                    message = "Error when trying to create Database. A database with the name {0} already exists.".format(
```

Comment:
Theoretically, IntegrityError could stand for a lot, not only for database name already exists. Maybe a more generic error message would be better

Reply:
This is code from the master, but makes sense, changed it nevertheless